locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+
+## Lifecycle of `.pipe()`
+
+To achieve a better understanding of backpressure, here is a flow-chart on the
+lifecycle of a [`Readable`][] stream being [piped][] into a [`Writable`][]
+stream:
+
+```javascript
++===============+               +===============+
+|   Your Data   |               |  .pipe(dest)  |
++===============+               |===============|
+        |                       |  Setup event  |
+        |                       |  closures:    |
++-------v-----------+           |               |         +-------------------+
+|  Readable Stream  >-----------+ * .data(cb)   +--------->  Writable Stream  |
++-^-------^-------^-+           | * .drain(cb)  |         +-------------------+
+  ^       ^       ^             | * .unpipe(cb) |                   |
+  |       ^       |             | * .error(cb)  |                   |
+  |       |       |             | * .finish(cb) |                   |
+  ^       |       |             | * .end(cb)    |           +=======v=========+
+  ^       |       ^             | * .close(cb)  |           |  .write(chunk)  |
+  |       |       ^             +---------------+           +=======+=========+
+  |       ^       |                                                 |
+  |       ^       |                              +------------------v---------+
+  ^       |       +-> if (!chunk)                |    Is this chunk too big?  |
+  ^       |       |     emit .end();             |    Is the queue busy?      |
+  |       |       +-> else                       +-------+----------------+---+
+  |       ^       |     emit .write();                   |                |
+  |       ^       ^                                   +--v---+        +---v---+
+  |       |       ^-----------------------------------<  No  |        |  Yes  |
+  ^       |                                           +------+        +---v---+
+  ^       |                                                               |
+  |       ^               emit .pause();          +=================+     |
+  |       ^---------------^-----------------------+  return false;  <-----+---+
+  |                                               +=================+         |
+  |                                                                           |
+  ^            when queue is empty     +============+                         |
+  ^------------^-----------------------<  Buffering |                         |
+               |                       |============|                         |
+               +> emit .drain();       |  ^Buffer^  |                         |
+               +> emit .resume();      +------------+                         |
+                                       |  ^Buffer^  |                         |
+                                       +------------+   add chunk to queue    |
+                                       |            <---^---------------------<
+                                       +============+
+```
```

Comment:
This diagram is making good progress! But I think you should split `pipe()` in its own step of the diagram, and not make it part of the "loop". The event handlers are, so we should show the distinction.

The syntax for events, e.g. `drain()`, is not common throughout the docs, so we should probaly use `on('drain', cb)`  instead.

also cc @lrlna who did some visualization on this in the past on paper.